cdap-tms/src/main/java/io/cdap/cdap/messaging/subscriber/AbstractEventSubscriberService.java

```diff
@@ -0,0 +1,160 @@
+package io.cdap.cdap.messaging.subscriber;
+
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import com.google.common.collect.AbstractIterator;
+import io.cdap.cdap.api.metrics.MetricsContext;
+import io.cdap.cdap.common.service.RetryStrategy;
+import io.cdap.cdap.common.utils.ImmutablePair;
+import io.cdap.cdap.common.utils.TimeBoundIterator;
+import io.cdap.cdap.proto.id.TopicId;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+import javax.annotation.Nullable;
+
+/**
+ * An abstract base class for implementing event deletion message consumption from TMS.
+ *
+ * @param <T> the type that each message will be decoded to.
+ */
+public abstract class AbstractEventSubscriberService<T> extends AbstractMessagingPollingService<T> {
```

Comment:
this looks like mostly duplicate code from AbstractMessagingSubscriberService, why do we need this?

Reply:
so had earlier discussed with @tivv and decided to have a separate abstract class that gets rid of additional methods like `loadMessageId` `storeMessageId` etc.  Instead of this, can we just extend AbstractMessagingSubscriberService and have no implementation for the unnecessary methods? @tivv

Reply:
The goal is to reuse in other children of AbstractMessagingPollingService. It seems that tons of code is duplicated